src/sentry/migrations/0024_auto_20191230_2052.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9.13 on 2019-12-30 20:52
+from __future__ import unicode_literals
+
+from datetime import timedelta, datetime
+
+from django.db import migrations
+
+
+def backfill_eventstream(apps, schema_editor):
+    """
+    Inserts Postgres events into the eventstream if there are recent events in Postgres.
+
+    This is for open source users migrating from 9.x who want to keep their events.
+    If there are no recent events in Postgres, skip the backfill.
+    """
+    from sentry import eventstore, eventstream
+    from sentry.utils.query import RangeQuerySetWrapper
+
+    Event = apps.get_model('sentry', 'Event')
+    Group = apps.get_model('sentry', 'Group')
+    Project = apps.get_model('sentry', 'Project')
+
+    days_to_check = 14
+
+    def get_events(last_days):
+        to_date = datetime.now()
+        from_date = to_date - timedelta(days=last_days)
+        return Event.objects.filter(datetime__gte=from_date, datetime__lte=to_date)
+
+    def _attach_related(_events):
+        project_ids = set([event.project_id for event in _events])
+        projects = {p.id: p for p in Project.objects.filter(id__in=project_ids)}
+        group_ids = set([event.group_id for event in _events])
+        groups = {g.id: g for g in Group.objects.filter(id__in=group_ids)}
+        for event in _events:
```

Comment:
At this point, we are iterating over `_events` 3 times. We can at least drop this to 2 by combining the creation of the `project_ids` and `group_ids` above in a single for loop (instead of using the handy comprehensions). I wonder if we can reduce this to a single loop and just querying the database more times or some neat Python trick.

Reply:
Ah, this was just a copy/paste of the backfill script, but I shall revisit this.